### PR TITLE
feat: Add extra detail to autosuggest onSelect handler

### PR DESCRIPTION
**Pull request by just-boris**
### Description

Follow up for https://github.com/cloudscape-design/components/pull/834. Another attempt to resolve AWSUI-20591, but this time in a backward compatible way

Related links, issue #, if available: AWSUI-20591

### How has this been tested?

Added extra tests 

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/autosuggest/__integ__/events-autosuggest.test.ts [hunk 1]
```diff
@@ -25,14 +25,15 @@ describe.each<boolean>([false, true])('Autosuggest events (expandToViewport=%s)'
   );
 
   test(
-    'should fire change event when selecting suggestions with SPACE',
+    'should allow entering spaces after focusing a dropdown item',
```
Comment (just-boris) [inline src/autosuggest/__integ__/events-autosuggest.test.ts:28]: The test logic did not change, but the title was misleading. Space key does not select any options (before and after this PR)

File: src/autosuggest/__tests__/autosuggest-and-token-group.test.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+const wrapper = createWrapper();
+
+function getSelectedTokens() {
+  return wrapper
+    .findTokenGroup()!
+    .findTokens()
+    .map(token => token.findLabel().getElement().textContent!.trim());
+}
+
+function Demo() {
```
Comment (just-boris) [inline src/autosuggest/__tests__/autosuggest-and-token-group.test.tsx:18]: Added a verbose test case to show a practical example where the `onSelect` handler can be used for good.

File: src/autosuggest/__tests__/autosuggest-and-token-group.test.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  wrapper.findAutosuggest()!.focus();
+  wrapper
+    .findAutosuggest()!
+    .findEnteredTextOption()!
+    .fireEvent(new MouseEvent('mouseup', { bubbles: true }));
+  expect(wrapper.findAutosuggest()!.findNativeInput().getElement()).toHaveValue('');
+  expect(getSelectedTokens()).toEqual(['custom']);
+});
+
+test('Removing tokens', () => {
```
Comment (gethinwebster) [inline src/autosuggest/__tests__/autosuggest-and-token-group.test.tsx:67]: [minor] is this testing something relevant to autosuggest? Compared to `"Adding tokens from the dropdown"` it's only really testing `Demo` rather than cloudscape components (unless I miss something)

File: src/autosuggest/interfaces.ts [hunk 4]
```diff
@@ -122,6 +122,7 @@ export namespace AutosuggestProps {
   export type StatusType = DropdownStatusProps.StatusType;
   export interface SelectDetail {
     value: string;
+    selectedOption?: Option;
```
Comment (gethinwebster) [inline src/autosuggest/interfaces.ts:125]: [minor] would it be worthwhile doing a dry run to live to check if any consumers are misusing these interfaces, so we can notify them in advance?

**Review by gethinwebster — APPROVED** [review]

File: src/autosuggest/__tests__/autosuggest-and-token-group.test.tsx (hunk 3 above)
Comment (just-boris) [reply src/autosuggest/__tests__/autosuggest-and-token-group.test.tsx:67]: This is not a pure autosuggest test suite, but also tests on the entire `<Demo>` component from this file to confirm that this entire use-case is doable
